### PR TITLE
feat: CRN-1122 Set Contentful Env for Serverless

### PR DESCRIPTION
**Pull request by peterstarling**
Notable changes:
* add a `setup` job before the `build`, `create-environment` and `deployment`
* user lowercase for pseudo-boolean values
* move the contentful env limit check to where the env is actually created

Outstanding tasks for another PR
* use contentful cli to create the env

### CODE REVIEW

**Review by m-doughty — CHANGES_REQUESTED** [review]
Due to the case change in the `checks` outputs (I don't have strong opinions on it either way), `on-remove.yml` will need to be changed where it checks.

I think there will also be changes to the guard clauses in the shell scripts (`create.sh` and `teardown.sh`).

File: .github/scripts/contentful/create.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-if [[ "ON_BRANCH_ENV" == "False" ]]; then
+if [[ "ON_BRANCH_ENV" == "false" ]]; then
```
Comment (markcsims) [inline .github/scripts/contentful/create.sh:3]:
```suggestion
if [[ "$ON_BRANCH_ENV" == "false" ]]; then
```

File: .github/scripts/contentful/create.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
   echo "No branch Environment required; not deleting environment."
```
Comment (markcsims) [inline .github/scripts/contentful/create.sh:4]: this message is confusing to me.

The variable is 'ON_BRANCH_ENV' i.e. whether we are on a dev branch. 

And this message is printed when we are not on dev.

Comment (m-doughty) [reply .github/scripts/contentful/create.sh:4]: Yeah agree, this should probably be something like "No branch environment to delete."

Comment (m-doughty) [reply .github/scripts/contentful/create.sh:4]: I have made this change @markcsims.

Comment (peterstarling) [inline .github/scripts/contentful/create.sh:4]:
```suggestion
  echo "No branch Environment required; not creating environment."
```

**Review by markcsims — APPROVED** [review]

**Pull request merged**